src/core/aws-net-websocket-protocol-rfc6455.adb

```diff
@@ -106,6 +113,57 @@ package body AWS.Net.WebSocket.Protocol.RFC6455 is
    function Is_Valid_Close_Code (Error : Error_Type) return Boolean;
    --  Returns True if the Error code is valid
 
+   function Create_Random_Mask return Masking_Key;
+   --  Create a random masking key
+
+   function Get_Websocket_Accept (Key : String) return String;
+   --  Compute the Sec-WebSocket-Accept header
+
+   -------------------------
+   -- Add_Connect_Headers --
+   -------------------------
+
+   overriding procedure Add_Connect_Headers
+      (Protocol : State;
+       URI      : String;
+       Headers  : in out AWS.Headers.List)
+   is
+      pragma Unreferenced (Protocol);
+      Ints : array (1 .. 4) of AWS.Utils.Random_Integer :=
+              (others => AWS.Utils.Random);
+      Hex : Stream_Element_Array (1 .. 16) with Import;
+      for Hex'Address use Ints'Address;
```

Comment:
Here too we could use an aspect BTW.

Reply:
And plenty of other places in AWS. I fixed the other cases in this file, but not in all the rest of the code

Reply:
Sure, thanks Manu. The AWS code pre-date the aspect support, from time to time I do some clean-up.